main.go

```diff
@@ -0,0 +1,423 @@
+			w.Write(errStr)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			return
+		}
+		resp, err := json.Marshal(responseTasksList{
+			Tasks: tasks,
+		})
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write(resp)
+		return
+	}
+	w.WriteHeader(http.StatusBadRequest)
+	w.Write([]byte("error: invalid method"))
+}
+func handlerTaskDone(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		id := req.URL.Query().Get("id")
+		if id == "" {
+			errStr, err := json.Marshal(struct {
+				Error string `json:"error"`
+			}{
+				Error: "не передан id",
+			})
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte(err.Error()))
+				return
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write(errStr)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			return
+		}
+		idInt, err := strconv.Atoi(id)
+		if err != nil {
+			errStr, err := json.Marshal(struct {
+				Error string `json:"error"`
+			}{
+				Error: "передан неверный id",
+			})
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte(err.Error()))
+				return
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write(errStr)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			return
+		}
+		task, err := task.GetById(idInt)
+		if err != nil {
+			errStr, err := json.Marshal(struct {
+				Error string `json:"error"`
+			}{
+				Error: err.Error(),
+			})
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte(err.Error()))
+				return
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write(errStr)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			return
+		}
+		err = task.Done()
+		if err != nil {
+			errStr, err := json.Marshal(struct {
+				Error string `json:"error"`
+			}{
+				Error: err.Error(),
+			})
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte(err.Error()))
+				return
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write(errStr)
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("{}"))
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		return
+	}
+	w.WriteHeader(http.StatusBadRequest)
+	w.Write([]byte("error: invalid method"))
+}
+
+func main() {
+	err := godotenv.Load()
+	if err != nil {
+		log.Fatalf("err loading: %v", err)
+		return
+	}
+
+	err = db.CheckDb()
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	webDir := "./web"
+	port, err := strconv.Atoi(os.Getenv("TODO_PORT"))
+	if err != nil {
+		port = tests.Port
+	}
+	http.Handle("/", http.FileServer(http.Dir(webDir)))
+	http.HandleFunc("/api/nextdate", authMiddleware(handlerApiNextDate))
```

Comment:
Иначе тесты не проходят
```suggestion
	http.HandleFunc("/api/nextdate", handlerApiNextDate)
```